pandas/core/window/indexers.py

```diff
@@ -120,3 +120,53 @@ def get_window_bounds(
             np.zeros(num_values, dtype=np.int64),
             np.arange(1, num_values + 1, dtype=np.int64),
         )
+
+
+class FixedForwardWindowIndexer(BaseIndexer):
+    """
+    Creates window boundaries for fixed-length windows that include the
+    current row.
+
+    Examples
+    --------
+    >>> df = pd.DataFrame({'B': [0, 1, 2, np.nan, 4]})
+    >>> df
+         B
+    0  0.0
+    1  1.0
+    2  2.0
+    3  NaN
+    4  4.0
+
+    >>> indexer = pd.api.indexers.FixedForwardWindowIndexer(window_size=2)
+    >>> df.rolling(window=indexer, min_periods=1).sum()
+         B
+    0  1.0
+    1  3.0
+    2  2.0
+    3  4.0
+    4  4.0
+    """
+
+    @Appender(get_window_bounds_doc)
+    def get_window_bounds(
+        self,
+        num_values: int = 0,
+        min_periods: Optional[int] = None,
+        center: Optional[bool] = None,
+        closed: Optional[str] = None,
+    ) -> Tuple[np.ndarray, np.ndarray]:
+
+        if center:
+            raise ValueError("Forward-looking windows can't have center=True")
+        if closed is not None:
+            raise ValueError(
+                "Forward-looking windows don't support setting the closed argument"
+            )
```

Comment:
@jreback
Implemented the error-raising behavior this way. Reasons below.
As far as I've been able to test, the arguments are being set by the `rolling` object in a general way and outside the control of the user. They are passed whether a particular `BaseIndexer` implementation needs them or not.
`min_periods` is getting set to `rolling.min_periods`. Same stuff for `center`: in my tests it got set to `False` by default. Only `closed` got passed as `None`.
I wrote the error-raising behavior with this in mind. We should raise if something that doesn't make sense got passed, but testing for None is too crude for our purposes.